code/mob/living/life/breath.dm

```diff
@@ -102,7 +102,11 @@
 			return // no breathing inside possessed objects
 		else if (istype(owner.loc, /obj/machinery/atmospherics/unary/cryo_cell))
 			return
-
+		else if (istype(owner.loc, /obj/machinery/bathtub) && owner.lying)
+			if (owner.loc.reagents.total_volume > 300)
+				var/obj/fluid/F = new
+				F.reagents = owner.loc.reagents
+				underwater = F
```

Comment:
I was going to review this like 2 days ago but I couldn't think of a clean way to do what you want here.
TL;DR this won't work- fluids track reagents via their `group`, rather than each fluid obj tracking their own reagents. This means that later on, when `force_mob_to_ingest` tries to make the occupant drink from the fluid they're drowning in, it'll see the fluid doesn't have a group and return without doing anything.

I suppose you could override the bathtub's `handle_internal_lifeform` to handle drowning "manually", but that would require some code duplication. grumble